workspaces/ui-v2/src/spectacle-implementations/cloud-viewer.tsx

```diff
@@ -107,7 +112,13 @@ export default function CloudViewer() {
             capturesService={data.opticContext.capturesService}
           >
             <BaseUrlProvider value={{ url: match.url }}>
-              <AnalyticsStore>
+              <AnalyticsStore
+                getMetadata={getMetadata(() =>
```

Comment:
There's something weird that I had to call `getMetadata(() => fn())` instead of `getMetadata(fn)` - where the first instance / render of this would return an invalid function...

Error I was getting `opticContext.configRepository.getApiName() -> `this.dependencies` is undefined` - which crashed the page

I didn't dig too much more since I think this is better replaced if we use redux + implement analytics middleware, but it is interesting.

Reply:
i think that's normal behavior if you pass a class method without rebinding it, but seems fine to wrap it anyway

Reply:
also passing a function to some hooks call it immediately, so you need to wrap, that might be it as well

Reply:
I was writing it as `getMetadata(data.opticContext.configRepository.getApiName)` which should have passed the function reference to the correct place. I think the first render is that it hasn't loaded yet (loading state) - and because i _think_ it can't tell the fn identity changes - it calls the old one. But wrapping this with an inline function means it's accessed on call?

It could also be that I'm using a useRef here. It's odd because the function is being called at the same time, it just had a reference to a stale instance and wrapping a fn means that we are able to access the correct function reference

Either way - this works 🤷